### DISCUSSION TITLE
Using submenus with icons end text

### DISCUSSION THREAD

**Discussion by AcernisSupport** [ant-design-blazor/ant-design-blazor#4127] [Q&A]
Good morning

I'm using SUBMENUS in the procedure I'm building but I saw that I want to combine ICON and TITLE I have to use this code

`@code
{
    RenderFragment stManagement =
    <span>@
        <Icon Type="appstore" Theme="outline"></Icon>
        <span>management</span>
    </span>;
}`

that I have seen in the examples on the site.

My problem is that within the procedure we use a resource file with the entries and read them with

(to inject it into the code) 
`@inject IStringLocalizer<Resources> Locale `

and we use it in this way (in this case a title in the MENUITEM)
`Title="@Locale["Common.Managements"]"`

I have done several tests (I have been fighting :) for three days now) to integrate IStringLocalizer into the RenderFragment code but I have not been able to understand how to do it.

Has anyone already had these problems and can help me with an example?

Thank you all and good work

**Comment by ElderJames** [comment]
Thanks for contacting us @AcernisSupport ,  you can use TitleTemplate like that:

```
            <SubMenu Title="Option 4">
                <TitleTemplate>
                    <Icon Type="usre" />  @localizer["Title"]
                </TitleTemplate>
                <ChildContent>
                    <MenuItem Key="setting4:1">Option 1</MenuItem>
                    <MenuItem Key="setting4:2">Option 2</MenuItem>
                </ChildContent>
            </SubMenu>
```
